metis-core/metis-core-service/src/test/java/eu/europeana/metis/core/service/TestDepublishRecordIdService.java

```diff
@@ -0,0 +1,153 @@
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import eu.europeana.metis.authentication.user.MetisUser;
+import eu.europeana.metis.core.dao.DepublishRecordIdDao;
+import eu.europeana.metis.core.dataset.DatasetExecutionInformation;
+import eu.europeana.metis.core.dataset.DatasetExecutionInformation.PublicationStatus;
+import eu.europeana.metis.core.rest.DepublishRecordIdView;
+import eu.europeana.metis.core.rest.ResponseListWrapper;
+import eu.europeana.metis.core.util.DepublishRecordIdSortField;
+import eu.europeana.metis.core.util.SortDirection;
+import eu.europeana.metis.core.utils.TestObjectFactory;
+import eu.europeana.metis.core.workflow.Workflow;
+import eu.europeana.metis.core.workflow.plugins.DepublishPluginMetadata;
+import eu.europeana.metis.exception.GenericMetisException;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class TestDepublishRecordIdService {
+
+  private static Authorizer authorizer;
+  private static OrchestratorService orchestratorService;
+  private static DepublishRecordIdDao depublishRecordIdDao;
+  private static DepublishRecordIdService depublishRecordIdService;
+  private static MetisUser metisUser;
+  private static String datasetId;
+
+  @BeforeAll
+  static void setUp() {
+
+    authorizer = mock(Authorizer.class);
+    orchestratorService = mock(OrchestratorService.class);
+    depublishRecordIdDao = mock(DepublishRecordIdDao.class);
+    metisUser = TestObjectFactory.createMetisUser(TestObjectFactory.EMAIL);
+    datasetId = Integer.toString(TestObjectFactory.DATASETID);
+
+    depublishRecordIdService = spy(new DepublishRecordIdService(authorizer, orchestratorService,
+        depublishRecordIdDao));
+
+  }
+
+  @BeforeEach
+  void cleanUp() {
+    reset(authorizer);
+    reset(orchestratorService);
+    reset(depublishRecordIdDao);
+    reset(depublishRecordIdService);
+  }
+
+  @Test
+  void addRecordIdsToBeDepublishedTest() throws GenericMetisException {
+    depublishRecordIdService.addRecordIdsToBeDepublished(metisUser, datasetId, "1002");
+
+    verify(authorizer, times(1)).authorizeWriteExistingDatasetById(metisUser, datasetId);
+    verify(depublishRecordIdService, times(1)).checkAndNormalizeRecordIds(any(), any());
+    verify(depublishRecordIdDao, times(1)).createRecordIdsToBeDepublished(any(), any());
```

Comment:
Perhaps you can check here not just that the method is called, but also that it's called with the right data? 
Same in the mathods below.

Reply:
Like we talked before, some methods (like `addRecordIdsToBeDepublished` and `deletePendingRecordIdsTest`) are mostly dependent on other objects and on their maethods, more specifically the object `DepublishRecordIdDao`, and those specific methods are already being tested on a different unit test class. So I believe in this case it is not necessary to check the values.
However, I have added these checks on the values on the other methods